frontend/packages/dev-console/src/components/pipelines/modals/triggers/TriggerBindingSelectorDropdown.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { FirehoseResult } from '@console/internal/components/utils';
+import { DropdownField } from '@console/shared';
+import { TriggerBindingKind } from '../../resource-types';
+
+type TriggerBindingSelectorProps = {
+  description?: string;
+  label: string;
+  onChange: (selectedTriggerBinding: TriggerBindingKind) => void;
+  triggerBindingData?: FirehoseResult<TriggerBindingKind[]>;
+};
+
+const TriggerBindingSelectorDropdown: React.FC<TriggerBindingSelectorProps> = (props) => {
+  const { description, label, onChange, triggerBindingData } = props;
+  const triggerBindings = triggerBindingData.data;
```

Comment:
we could do this here and use the collected item down for readability.
```suggestion
  const triggerBindings = triggerBindingData.data.reduce(
        (acc, triggerBinding) => ({
          ...acc,
          [triggerBinding.metadata.name]: triggerBinding.metadata.name,
        }),
        {},
      );
```